lib/errors/error_handling.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/errors/course_update_error_handling"
+
+module ErrorHandling
+  include CourseUpdateErrorHandling
+
+  def perform_error_handling_tasks(error, level, extra, course, optional_params)
+    return report_exception_sentry(error, level, extra) unless course.present?
+    sentry_tag_uuid = SecureRandom.uuid
```

Comment:
I suggest using a uuid for every error, and then only doing the save_course_error_record if if the course is present.

Reply:
I have done something different, please have a look at the `def perform_error_handling` in `ErrorHandling` and `def perform_course_error_handling` in `CourseUpdateErrorHandling`. What I have done is actually momved all the part relating to `uuid` in `CourseUpdateErrorHandling` itself so that the control flow in the methods is not confusing as well as in the relevant modules.